sagemaker_neo_compilation_jobs/mxnet_mnist/mnist.py

```diff
@@ -107,42 +105,63 @@ def parse_args():
 
     return parser.parse_args()
 
-### NOTE: this function cannot use MXNet
-def neo_preprocess(payload, content_type):
+### NOTE: model_fn and transform_fn are used to load the model and serve inference
+def model_fn(model_dir):
+    import mxnet as mx
+    import neomxnet  # noqa: F401
     import logging
+    from collections import namedtuple
+    logging.info('Invoking user-defined model_fn')
+    # change context to mx.gpu() when optimizing and depolying with Neo for GPU endpoints
+    ctx = mx.cpu()
+    Batch = namedtuple('Batch', ['data'])
+    sym, arg_params, aux_params = mx.model.load_checkpoint(os.path.join(model_dir, 'compiled'), 0)
+    mod = mx.mod.Module(symbol=sym, context=ctx, label_names=None)
+    exe = mod.bind(for_training=False,
+                   data_shapes=[('data', (1,784))],
+                   label_shapes=mod._label_shapes)
+    mod.set_params(arg_params, aux_params, allow_missing=True)
+    # run warm-up inference on empty data
+    data = mx.nd.empty((1,784), ctx=ctx)
+    mod.forward(Batch([data]))
+    return mod
+
+def transform_fn(mod, payload, input_content_type, output_content_type):
     import numpy as np
+    import mxnet as mx
+    import neomxnet  # noqa: F401
```

Comment:
do we need this here too? I think having it in `model_fn` is enough

Reply:
refactored and tested it again

Reply:
What was your conclusion? We shouldnt need `import neomxnet` in both `model_fn` and `transform_fn` since loading it once in the same python process (in the `model_fn`) should keep it in Python's cache when the `transform_fn` is executed. Heres a reference: https://stackoverflow.com/a/3095167

Reply:
Agreed. I have made changes after testing